tests/global_communication/test_mpi.cpp

```diff
@@ -117,25 +110,24 @@ TEST(mpi, gather_string) {
 
     auto s = make_string(id);
 
-    auto gathered = mpi::gather(s, 0);
+    auto gathered = mpi::gather(s, 0, MPI_COMM_WORLD);
 
     if (!id) {
-        ASSERT_TRUE(policy::size()==(int)gathered.size());
+        ASSERT_TRUE(g_context.size()==(int)gathered.size());
         for (std::size_t i=0; i<gathered.size(); ++i) {
             EXPECT_EQ(make_string(i), gathered[i]);
         }
     }
 }
 
 TEST(mpi, gather) {
-    using policy = mpi_global_policy;
-
-    int id = policy::id();
+    mpi_context ctx(MPI_COMM_WORLD);
+    int id = ctx.id();
```

Comment:
This sort of mixes the `mpi` functionality tests with `mpi_context`. Is it possible to split the tests so that `test_mpi` just checks the MPI wrappers, and a different set of tests e.g. `test_mpi_context` tests the `distributed_context` implementation?

Reply:
I removed all `g_context` references from this file. The global context was being used soley to get the rank and number of mpi ranks, so I replaced those calls with `mpi::rank` and `mpi::size`.

There should be seperate unit tests for the `distributed_context` (there were none before, and there are still none now). However it isn't a super high priority, because `mpi_context` is a very thin wrapper around these calls, so these are the key ones to test).